### PR TITLE
Add RV-8263-C8 driver

### LINKED ISSUE #267: Improve RV-8263-C8 driver in Zephyr
- Alarm functions missing
- Needed for https://github.com/jakkra/ZSWatch-HW/issues/44

### LINKED ISSUE #281: Move compass app configuration into "Applications" menu
Not that important

image

### LINKED ISSUE #284: Funkytown_song not used
https://github.com/jakkra/ZSWatch/blob/8152f4b07cb735c87b3477adf24a910605150c7b/app/src/drivers/zsw_buzzer.c#L74

### LINKED ISSUE #285: Disable touch for battery and step counter spinner in digital watchface

### LINKED ISSUE #309: Using the RTC update for the 1s periodic event
Replace the timer for the 1s periodic callback with the RTC update function when using rev 5

image

### LINKED ISSUE #311: Disable UART in DT
Disable UART for rev >3

```
&uart0 {
    status = "disabled";
};
```

### PR DESCRIPTION
**Pull request by Kampi**
- Add RV-8263-C8 driver
- Some smaller fixes

Closes #284 
Closes #267 
Closes #285 
Closes #281 
Closes #309
Closes #311

### CODE REVIEW

**Comment by jakkra** [comment]
Skimmed through, see nothing that looks out of the ordinary. Let's see if it works later :)

**Comment by Kampi** [comment]
@jakkra I dropped my driver and use the driver from Zephyr instead. The driver is functional on HW (tested with dev-board) and also tested via the Zephyr unit test for the RTC API.

**Comment by Kampi** [comment]
@jakkra have you tested it on rev 5? I did some smaller changes on the driver so I have to update the patch.

**Comment by jakkra** [comment]
> @jakkra have you tested it on rev 5? I did some smaller changes on the driver so I have to update the patch.

Nope, I'll let you have the honor 😉

File: app/patches/rv8263_rtc.patch [hunk 1]
```diff
@@ -0,0 +1,950 @@
...
++	.get_calibration = rv8263c8_calibration_get,
++#endif
++};
++
++#define RV8263_DEFINE(inst)                                                                        \
++	static struct rv8263c8_data rv8263c8_data_##inst;                                          \
++	static const struct rv8263c8_config rv8263c8_config_##inst = {                             \
++		.i2c_bus = I2C_DT_SPEC_INST_GET(inst),                                             \
++		.clkout = DT_INST_ENUM_IDX(inst, clkout),                                          \
++		IF_ENABLED(DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios),                            \
```
Comment (Kampi) [inline app/patches/rv8263_rtc.patch:907]: @jakkra you can apply the complete patch but this line causes issues with the compilation (`DT_ANY_INST_HAS_PROP_STATUS_OKAY`). A "dirty" solution would be to replace `DT_ANY_INST_HAS_PROP_STATUS_OKAY` with `1`. :D

**Comment by Kampi** [comment]
@jakkra let´s merge this repo. The RTC driver has been added, but no RTC in the code yet.

**Pull request merged**